comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go

```diff
@@ -212,6 +216,24 @@ func createNewAutoConfig(schedulerController *scheduler.Controller, secretResolv
 		filterStore:              filterStore,
 		telemetryStore:           acTelemetry.NewStore(telemetryComp),
 	}
+
+	secretResolver.SubscribeToChanges(func(_, origin string, _ []string, oldValue, _ any) {
+		oldValueStr, ok := oldValue.(string)
+		if !ok {
+			return
+		}
+
+		isEnc, _ := utils.IsEnc(oldValueStr)
+		// - An empty old value means this secret was initially resolved and isn't a refresh.
+		// - An unresolved ([ENC]) value implies this secret was triggered by a cache hit, not a refresh.
+		if oldValueStr == "" || isEnc {
```

Comment:
Some callbacks are triggered by cache hit or initial secret resolution (which aren't refreshes), but there's no explicit way of knowing that without doing checks like this. 

These callbacks should mostly be from non-autodiscovery secrets because autodiscovery disables `notify` when calling `secretsResolver.Resolve()` but I've added this here as a precaution in case of future changes.